servers/rendering/renderer_rd/renderer_canvas_render_rd.cpp

```diff
@@ -1627,6 +1641,38 @@ RendererCanvasRenderRD::RendererCanvasRenderRD() {
 
 		default_samplers.default_filter = RS::CANVAS_ITEM_TEXTURE_FILTER_LINEAR;
 		default_samplers.default_repeat = RS::CANVAS_ITEM_TEXTURE_REPEAT_DISABLED;
+		Vector<RID> samplers = material_storage->samplers_rd_get_default().get_rids();
+		for (int i = 0; i < DEFAULT_MATERIAL_SAMPLER_COUNT; i++) {
+			default_material_samplers[i] = samplers[i];
+		}
+	}
+
+	uint32_t push_pad_size = 0;
+
+	{
+		// determine max number of texture slots we can use, which is a function of two driver limits:
```

Comment:
This block determines the maximum number of textures we can use, which as the comment describes, is a function of the texture slots and push constant size. We dynamically determine the push constant size based on these limits to maximise the batching for a given driver.